pkg/apis/controlplane/v1beta1/types.go

```diff
@@ -73,25 +73,17 @@ type ExternalEntityReference struct {
 	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,opt,name=namespace"`
 }
 
-// Endpoint represents an external endpoint.
-type Endpoint struct {
-	// IP is the IP address of the Endpoint.
-	IP IPAddress `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
-	// Ports is the list NamedPort of the Endpoint.
-	Ports []NamedPort `json:"ports,omitempty" protobuf:"bytes,2,rep,name=ports"`
-}
-
 // GroupMember represents resource member to be populated in Groups.
 // This supersedes GroupMemberPod, and will eventually replace it.
 type GroupMember struct {
 	// Pod maintains the reference to the Pod.
 	Pod *PodReference `json:"pod,omitempty" protobuf:"bytes,1,opt,name=pod"`
-
 	// ExternalEntity maintains the reference to the ExternalEntity.
 	ExternalEntity *ExternalEntityReference `json:"externalEntity,omitempty" protobuf:"bytes,2,opt,name=externalEntity"`
-
-	// Endpoints maintains a list of EndPoints associated with this groupMember.
-	Endpoints []Endpoint `json:"endpoints,omitempty" protobuf:"bytes,3,rep,name=endpoints"`
+	// IP is the IP address of the Endpoints associated with the GroupMember.
+	IPs []IPAddress `json:"ips,omitempty" protobuf:"bytes,3,rep,name=ips"`
+	// Ports is the list NamedPort of the GroupMember.
+	Ports []NamedPort `json:"ports,omitempty" protobuf:"bytes,4,rep,name=ports"`
```

Comment:
I think this is not a API compatible change, I assume this doesn't break upgrade test because `GroupMember` is currently not used in antrea itself, right?

Reply:
Yes GroupMember is only used for externalEntities as of now.

Reply:
In order to start abiding by the rules in https://github.com/vmware-tanzu/antrea/pull/1352, I believe we should consider using a `v1beta2` version of the API with the necessary conversion functions between internal version and . We will keep serving `v1beta1` for 9 months since it is beta. @Dyanngg is investigating this. Even if it may not be strictly necessary today because of the limited number of consumers of this API, this is a good habit to take IMO and a good exercise.